sycl/source/detail/context_impl.cpp

```diff
@@ -116,6 +116,14 @@ KernelProgramCache &context_impl::getKernelProgramCache() const {
   return MKernelProgramCache;
 }
 
+bool
+context_impl::hasDevice(std::shared_ptr<detail::device_impl> Device) const {
+  for (auto D : MDevices)
```

Comment:
This looks like an STL algorithm. OK, not very compelling while we do not have yet the C++20 syntax here...